libcontainer/standard_init_linux.go

```diff
@@ -179,6 +185,20 @@ func (l *linuxStandardInit) Init() error {
 	if err != nil {
 		return err
 	}
+	// Set seccomp as close to execve as possible, so as few syscalls take
+	// place afterward (reducing the amount of syscalls that users need to
+	// enable in their seccomp profiles).
+	if l.config.Config.Seccomp != nil && l.config.NoNewPrivileges {
+		seccompFd, err := seccomp.InitSeccomp(l.config.Config.Seccomp)
+		if err != nil {
+			return newSystemErrorWithCause(err, "init seccomp")
+		}
+		if seccompFd != -1 {
+			if err := syncParentSeccompHooks(l.pipe, seccompFd); err != nil {
+				return errors.Wrap(err, "seccomp hooks")
+			}
```

Comment:
> However, I think that we should close it explicitly after `syncParentSeccompHooks` (after the seccomp fd has been picked up by the parent process).
> 
> ```
> unix.Close(seccompFd)
> ```

I agree. 

> Otherwise, the processes created by the startContainer hook inside the container cannot get the seccomp fd with pidfd_getfd on pid1.
> 

I don't get this point. What is the startContainer hook relation with this? Is it a use case where the hook can get the seccomp fd using `pidfd_getfd`?

Reply:
I meant to block the hypothetical attack scenario where the attacker could install a startContainer hook. As far as I understand, this hook is executed in the container namespaces at a time where pid1 is still runc-init, with the seccomp fd open. So the attacker could fetch /proc/1/fd/42.